### PR TITLE
HexEditor+LibGUI: Initial version of HexEditor, LibGUI bug fix

### PR DESCRIPTION
**Pull request by xeons**
Initial version of Hex Editor

I was encountering an entire system crash when the window server attempted to do something with the shortcut text on a submenu. This bug only seemed to appear when I had a lone submenu inside of a menu.

Also fixed a bug with the save dialog involving references.

### CODE REVIEW

File: Libraries/LibGUI/GMenu.cpp [hunk 1]
```diff
@@ -113,6 +113,11 @@ int GMenu::realize_menu()
             request.menu.enabled = true;
             request.menu.checkable = false;
             request.menu.checked = false;
+            
+            // no shortcut on submenu, make sure this is cleared out
```
Comment (awesomekling) [inline Libraries/LibGUI/GMenu.cpp]: This looks like a useful fix, but please make it in a separate commit.

File: Libraries/LibGUI/GFilePicker.cpp [hunk 2]
```diff
@@ -30,10 +30,10 @@ Optional<String> GFilePicker::get_open_filepath()
 
 Optional<String> GFilePicker::get_save_filepath(const String& title, const String& extension)
 {
-    GFilePicker picker(Mode::Save, String::format("%s.%s", title.characters(), extension.characters()));
+    auto picker = GFilePicker::construct(Mode::Save, String::format("%s.%s", title.characters(), extension.characters()));
```
Comment (awesomekling) [inline Libraries/LibGUI/GFilePicker.cpp]: This is already fixed in LibGUI, please rebase on top of master. :)

File: Applications/HexEditor/GHexEditor.h [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <AK/Function.h>
+#include <AK/HashMap.h>
+#include <AK/NonnullOwnPtrVector.h>
+#include <AK/NonnullRefPtrVector.h>
+#include <AK/StdLibExtras.h>
+#include <LibDraw/TextAlignment.h>
+#include <LibGUI/GScrollableWidget.h>
+
+class GHexEditor : public GScrollableWidget {
```
Comment (awesomekling) [inline Applications/HexEditor/GHexEditor.h]: The `G` prefix is generally used for classes in LibGUI. Since this class is outside LibGUI, you can just call it `HexEditor` or something like that instead.

File: Libraries/LibGUI/GFilePicker.cpp (hunk 2 above)
Comment (xeons) [reply Libraries/LibGUI/GFilePicker.cpp]: Whoops, fixed now.

File: Libraries/LibGUI/GMenu.cpp (hunk 1 above)
Comment (xeons) [reply Libraries/LibGUI/GMenu.cpp]: Got this broken out now.

**Comment by awesomekling** [comment]
Delightful :smile:
